awx/ui/client/src/shared/stateDefinitions.factory.js

```diff
@@ -584,27 +584,97 @@ function($injector, $stateExtender, $log, i18n) {
                     squashSearchUrl: true,
                     url: '/add-user',
                     params: {
-                        user_search: {
+                        add_user_search: {
                             value: { order_by: 'username', page_size: '5' },
-                            dynamic: true,
+                            dynamic: true
                         }
                     },
                     views: {
                         [`modal@${formStateDefinition.name}`]: {
-                            template: `<add-rbac-resource users-dataset="$resolve.usersDataset" selected="allSelected" resource-data="$resolve.resourceData" without-team-permissions="true" title="` + i18n._('Add Users') + `"></add-rbac-resource>`
+                            template: `<add-rbac-resource default-params="$resolve.defaultParams" users-dataset="$resolve.usersDataset" selected="allSelected" resource-data="$resolve.resourceData" without-team-permissions="true" title="` + i18n._('Add Users') + `"></add-rbac-resource>`
                         }
                     },
                     ncyBreadcrumb:{
                         skip:true
                     },
                     resolve: {
-                        usersDataset: ['addPermissionsUsersList', 'QuerySet', '$stateParams', 'GetBasePath',
-                            function(list, qs, $stateParams, GetBasePath) {
+                        orgId:  ['$stateParams', 'Rest', 'GetBasePath', function($stateParams, Rest, GetBasePath) {
+                            let id;
+                            if ($stateParams.team_id) {
+                                Rest.setUrl(GetBasePath('teams') + `${$stateParams.team_id}`);
+                                id = Rest.get().then(({data}) => {
+                                    return data.summary_fields.organization.id;
+                                });
+                            } else {
+                                id = null;
+                            }
+                            return id;
+                        }],
+                        teamRoles:  ['$stateParams', 'Rest', 'GetBasePath', 'i18n', function($stateParams, Rest, GetBasePath, i18n) {
+                            let roles = null;
+                            if ($stateParams.team_id) {
+                                const basePath = GetBasePath('teams') + `${$stateParams.team_id}/object_roles`;
+                                Rest.setUrl(basePath);
+                                roles = Rest.get().then(({data}) => {
+                                    return data.results
+                                        .filter(({name}) => name === i18n._('Member') || name === i18n._('Admin'))
+                                        .map(({id}) => id)
+                                        .join(',');
```

Comment:
Suggestion: 
- The various ways in which we are filtering, mapping, and joining the returned values from the API in order to craft a return string of comma separated IDs to pass as our queryParams could be re-written as two functions that could handle any number of role names we pass it and format it in the way that we expect that could handle any changes in the future with how query strings are done:

```
const join = (arr) => arr.join(',');
const filterOut = (roles, criteria) => {
    return roles.reduce((filtered, {name, id}) => {
        criteria.forEach(filter => {
            if (name === i18n._(`${filter}`)) {
                filtered.push(id)
            }
        })
        return filtered;
    }, []);
}
```

This can then be used like so:

```
role = Rest.get().then(({data}) => {
    return join(filterOut(data.results, ['A', 'B', 'C']));
});
```